SQL: Add method args to PERCENTILE/PERCENTILE_RANK (#65026)
elastic/elasticsearch: docs/reference/sql/functions/aggs.asciidoc

[source, sql]
--------------------------------------------------
PERCENTILE(
    field_name,         <1>
    percentile[,        <2>
    method[,            <3>
    method_parameter]]) <4>
--------------------------------------------------

*Input*:

<1> a numeric field
<2> a numeric literal

> <3> optional string literal to indicate the <<search-aggregations-metrics-percentile-aggregation-approximation,percentile algorithm>>. Possible values: `tdigest` or `hdr`. Defaults to `tdigest`.

Reviewer: Current copy is fine as is, but this is a little more clear.
```suggestion
<3> optional string literal for the
<<search-aggregations-metrics-percentile-aggregation-approximation,percentile
algorithm>>. Possible values: `tdigest` or `hdr`. Defaults to `tdigest`.
```

> <4> optional numeric literal for configuring the <<search-aggregations-metrics-percentile-aggregation-approximation,percentile algorithm>>. Configures `compression` for `tdigest` or `number_of_significant_value_digits` for `hdr`. The default is the same as that of the backing algorithm.

Reviewer: Based on Andrei's review, `integer` may be a good replacement for `numeric literal`? I'd also streamline the wording slightly and remove the repeated link.
```suggestion
<4> optional integer that configures the percentile algorithm.
Configures `compression` for `tdigest` or `number_of_significant_value_digits`
for `hdr`. Uses the same defaults as the percentile algorithm.
```

> <4> optional numeric literal for configuring the <<search-aggregations-metrics-percentile-aggregation-approximation,percentile algorithm>>. Configures `compression` for `tdigest` or `number_of_significant_value_digits` for `hdr`. The default is the same as that of the backing algorithm.

Author: Same as above, technically `<3>` and `<4>` can be expressions too.

include-tagged::{sql-specs}/docs/docs.csv-spec[aggPercentileScalars]
--------------------------------------------------

["source","sql",subs="attributes,macros"]
--------------------------------------------------
include-tagged::{sql-specs}/docs/docs.csv-spec[aggPercentileWithPercentileConfig]
--------------------------------------------------

[[sql-functions-aggs-percentile-rank]]
==== `PERCENTILE_RANK`

.Synopsis:
[source, sql]
--------------------------------------------------
PERCENTILE_RANK(
    field_name,         <1>
    value[,             <2>
    method[,            <3>
    method_parameter]]) <4>
--------------------------------------------------

*Input*:

<1> a numeric field
<2> a numeric literal

> <3> optional string literal to indicate the <<search-aggregations-metrics-percentile-aggregation-approximation,percentile algorithm>>. Possible values: `tdigest` or `hdr`. Defaults to `tdigest`.

Reviewer: Same as above.
```suggestion
<3> optional string literal for the
<<search-aggregations-metrics-percentile-aggregation-approximation,percentile
algorithm>>. Possible values: `tdigest` or `hdr`. Defaults to `tdigest`.
```

> <4> optional numeric literal for configuring the <<search-aggregations-metrics-percentile-aggregation-approximation,percentile algorithm>>. Configures `compression` for `tdigest` or `number_of_significant_value_digits` for `hdr`. The default is the same as that of the backing algorithm.

Reviewer: Same as above.
```suggestion
<4> optional integer that configures the percentile algorithm.
Configures `compression` for `tdigest` or `number_of_significant_value_digits`
for `hdr`. Uses the same defaults as the percentile algorithm.
```
